api/task_manager.cc

```diff
@@ -0,0 +1,172 @@
+#include <seastar/core/coroutine.hh>
+
+#include "task_manager.hh"
+#include "api/api-doc/task_manager.json.hh"
+#include "db/system_keyspace.hh"
+#include "column_family.hh"
+#include "unimplemented.hh"
+#include "storage_service.hh"
+
+#include <utility>
+
+namespace api {
+
+namespace tm = httpd::task_manager_json;
+using namespace json;
+
+inline bool filter_tasks(seastar::shared_ptr<task_manager::task> task, std::unordered_map<sstring, sstring>& query_params) {
+    return (!query_params.contains("module") || query_params["module"] == task->get_status().type) &&
+        (!query_params.contains("keyspace") || query_params["keyspace"] == task->get_status().keyspace) &&
+        (!query_params.contains("table") || query_params["table"] == task->get_status().table);
+}
+
+tm::task_status make_status(task_manager::task_status status) {
+    auto start_time = gc_clock::to_time_t(status.start_time);
+    auto end_time = gc_clock::to_time_t(status.end_time);
+    ::tm st, et;
+    ::gmtime_r(&end_time, &et);
+    ::gmtime_r(&start_time, &st);
```

Comment:
I am not sure which time you mean. I think that `st` and `et` need to be of type `::tm` (`json::date_time`) since it is required by `task_manager.json.hh` generated file.

Reply:
@denesb see https://github.com/scylladb/scylladb/blob/ba42852b0e1d2bec677e60a863c505a637884e04/api/storage_service.cc#L1190

Reply:
We're converting a gc_clock::time_point to json::datetime, that needs a `struct tm` as input.

Reply:
We could return a timeuuid in theory but let's leave it to the virtual table